hadoop-hdds/common/src/main/java/org/apache/hadoop/utils/db/RDBTable.java

```diff
@@ -122,7 +122,11 @@ public boolean isEmpty() throws IOException {
   @Override
   public boolean isExist(byte[] key) throws IOException {
     try {
-      return db.get(handle, key) != null;
+      // RocksDB#keyMayExist
+      // If the key definitely does not exist in the database, then this
+      // method returns false, else true.
+      return db.keyMayExist(handle, key, new StringBuilder())
```

Comment:
As discussed offline, can we add some tests with proto objects and see will it cause any issues.

As our data in DB is byteArray of proto-objects.